### PR TITLE
Update michelin phy scripts

### PR DESCRIPTION
**Pull request by daviddanan**
Update Michelin-related scripts:

-General refactoring for getfem related scripts
-Extension to 3D for physical solver (mesh generation, static/quasi-static problems, contact/rolling etc...)
-Minor operation for datasets
-Repair notebooks for Benchmark 1 & 2

Note: there is an extra dependency with the following library 
https://gitlab.com/drti/basic-tools
that can be installed with conda (https://anaconda.org/drti/basictools). 
For now, it is not used a lot but it might in the near future, it is quite useful.

### CODE REVIEW

File: lips/dataset/powergridDataSet.py [hunk 1]
```diff
@@ -599,3 +596,20 @@ def reconstruct_output(self, data: "np.ndarray") -> dict:
             predictions[attr_nm] = data[:, prev_:(prev_ + this_var_size)]
             prev_ += this_var_size
         return predictions
+
+    def __getitem__(self, item:int)->tuple:
+        currentX = {xVarName:self.data[inputName][item] for xVarName in self._attr_x}
+        currentTau = {tauVarName:self.data[tauName][item] for tauVarName in self._attr_tau}
+        currentY = {yVarName:self.data[outputName][item] for yVarName in self._attr_y}
+        return (currentX,currentTau),currentY
```
Comment (Mleyliabadi) [inline lips/dataset/powergridDataSet.py:604]: I have already a function to do a similar task at : https://github.com/Mleyliabadi/LIPS/blob/UpdateMichelinPhyScripts/lips/dataset/powergridDataSet.py#L497

File: lips/dataset/powergridDataSet.py [hunk 2]
```diff
@@ -599,3 +596,20 @@ def reconstruct_output(self, data: "np.ndarray") -> dict:
...
+from lips import GetRootPath
+
+if __name__ == '__main__':
+    #Can not load dataset without grid2op (not required apparently, todo)
+    configFilePath=GetRootPath()+os.path.join("..","tests","data","powergrid","l2rpn_case14_sandbox","Benchmark1","train")
+    CONFIG_PATH_BENCHMARK=GetRootPath()+os.path.join("..","configurations","powergrid","benchmarks","l2rpn_case14_sandbox.ini")
+
+    myConfigManager=ConfigManager(CONFIG_PATH_BENCHMARK)
+    myDataset=PowerGridDataSet(config=myConfigManager,
+                 name="train")
```
Comment (Mleyliabadi) [inline lips/dataset/powergridDataSet.py:615]: Why I should have these lines in my class definition ?

File: lips/augmented_simulators/tensorflow_models/fully_connected.py [hunk 3]
```diff
@@ -48,10 +48,10 @@ class TfFullyConnected(TensorflowSimulator):
     """
     def __init__(self,
                  sim_config_path: str,
+                 bench_config_path: Union[str, pathlib.Path],
```
Comment (Mleyliabadi) [inline lips/augmented_simulators/tensorflow_models/fully_connected.py:51]: It is better to have `bench_config_name` near to `bench_config_path` which are related arguments

File: lips/augmented_simulators/tensorflow_models/leap_net.py [hunk 4]
```diff
@@ -67,10 +67,10 @@ class LeapNet(TensorflowSimulator):
     """
     def __init__(self,
                  sim_config_path: str,
+                 bench_config_path: Union[str, pathlib.Path],
```
Comment (Mleyliabadi) [inline lips/augmented_simulators/tensorflow_models/leap_net.py:70]: It is better to have `bench_config_name` near to `bench_config_path` which are related arguments

File: lips/augmented_simulators/torch_models/fully_connected.py [hunk 5]
```diff
@@ -47,10 +47,10 @@ class TorchFullyConnected(nn.Module):
     """
     def __init__(self,
                  sim_config_path: Union[pathlib.Path, str],
+                 bench_config_path: Union[str, pathlib.Path],
```
Comment (Mleyliabadi) [inline lips/augmented_simulators/torch_models/fully_connected.py:50]: It is better to have bench_config_name near to bench_config_path which are related arguments

File: lips/augmented_simulators/torch_models/u_net.py [hunk 6]
```diff
@@ -189,12 +189,12 @@ class TorchUnet(nn.Module):
         log path
     """
     def __init__(self,
+                 sim_config_path: str,
+                 bench_config_path: Union[str, pathlib.Path],
```
Comment (Mleyliabadi) [inline lips/augmented_simulators/torch_models/u_net.py:193]: It is better to have bench_config_name near to bench_config_path which are related arguments

**Review by Mleyliabadi — CHANGES_REQUESTED** [review]
Some minor changes may be performed before merging the pull request

File: lips/dataset/powergridDataSet.py (hunk 1 above)
Comment (Mleyliabadi) [reply lips/dataset/powergridDataSet.py]: However, it can be useful to keep it for the sake of homogeneity through the other dataset classes.

Comment (daviddanan) [reply lips/dataset/powergridDataSet.py]: Ah, i see. 
In that case we could for instance properly define __getitem__ in the base class, the method would call getData.
On it!

File: lips/dataset/powergridDataSet.py (hunk 2 above)
Comment (daviddanan) [reply lips/dataset/powergridDataSet.py]: Fair enough, removed.
It will become an issue instead.

**Review by Mleyliabadi — APPROVED** [review]
Good to go

**Pull request merged**